### PR TITLE
stateMachine does now save only absolute values

### PR DESCRIPTION
**Pull request by a19chrel**
stateMachine does now save only absolute values.

### CODE REVIEW

File: DuggaSys/diagram.js [hunk 1]
```diff
@@ -470,8 +481,17 @@ class StateMachine
         } else {
             console.error("Passed invalid argument to StateMachine.save() method. Must be a StateChange object!");
         }
+    };
+        settings.replay.timestamps = { 0: 0 }; // Clear the array with all timestamp.
+        this.historyLog.forEach(historyEntry => {
+            // settings.replay.timestamps[Object.keys(settings.replay.timestamps[Object.keys(settings.replay.timestamps).length - 1])]
+
+            if (settings.replay.timestamps[Object.keys(settings.replay.timestamps)[Object.keys(settings.replay.timestamps).length-1]] != historyEntry.time) {
```
Comment (a19hammi) [inline DuggaSys/diagram.js:489]: Please break this down.

File: DuggaSys/diagram.js [hunk 2]
```diff
@@ -470,8 +481,17 @@ class StateMachine
         } else {
             console.error("Passed invalid argument to StateMachine.save() method. Must be a StateChange object!");
         }
+    };
+        settings.replay.timestamps = { 0: 0 }; // Clear the array with all timestamp.
+        this.historyLog.forEach(historyEntry => {
+            // settings.replay.timestamps[Object.keys(settings.replay.timestamps[Object.keys(settings.replay.timestamps).length - 1])]
```
Comment (a19hammi) [inline DuggaSys/diagram.js:487]: Old commented code during testing?

File: DuggaSys/diagram.js [hunk 3]
```diff
@@ -488,25 +508,38 @@ class StateMachine
...
+            // Increase the currentHistoryIndex by one
+            this.currentHistoryIndex++;
 
-        // Restore the state
-        this.restoreState(this.historyLog[this.currentHistoryIndex]);
+            // Restore the state
+            this.restoreState(this.historyLog[this.currentHistoryIndex]);
+
+            var doNextState = false;
+            if (this.historyLog[this.currentHistoryIndex + 1]) doNextState = (this.historyLog[this.currentHistoryIndex].time == this.historyLog[this.currentHistoryIndex + 1].time)
```
Comment (a19hammi) [inline DuggaSys/diagram.js:541]: Break this down

**Review by a19hammi — CHANGES_REQUESTED** [review]

File: DuggaSys/diagram.js (hunk 1 above)
Comment (a19chrel) [reply DuggaSys/diagram.js]: The one-liner has now been split apart to make it more readable.

File: DuggaSys/diagram.js (hunk 2 above)
Comment (a19chrel) [reply DuggaSys/diagram.js]: Nice catch! It has now been removed.

File: DuggaSys/diagram.js (hunk 3 above)
Comment (a19chrel) [reply DuggaSys/diagram.js]: The selection is no longer a one-liner, bracket has been added to make it more readable.

**Review by a19aleda — APPROVED** [review]
After the change of the one-liners, this looks good imo :)

**Review by a19hammi — APPROVED** [review]

**Review by a19hammi — COMMENTED** [review]
`DuggaSys/hej` ???

**Pull request merged**